x-pack/plugin/mapper-aggregate-metric/src/main/java/org/elasticsearch/xpack/aggregatemetric/mapper/AggregateDoubleMetricFieldMapper.java

```diff
@@ -494,18 +494,7 @@ public BucketedSort newBucketedSort(
 
         @Override
         public ValueFetcher valueFetcher(SearchExecutionContext context, String format) {
-            if (format != null) {
-                throw new IllegalArgumentException("Field [" + name() + "] of type [" + typeName() + "] doesn't support formats.");
-            }
-
-            return new SourceValueFetcher(name(), context) {
-                @Override
-                @SuppressWarnings("unchecked")
-                protected Object parseSourceValue(Object value) {
-                    Map<String, Double> metrics = (Map<String, Double>) value;
-                    return metrics.get(defaultMetric.name());
-                }
-            };
+            return SourceValueFetcher.identity(name(), context, format);
```

Comment:
Should we require that format is null for this field type? 
@romseygeek can you help here?

Reply:
I just noticed  [`FieldValueFetcher#identity` method](https://github.com/elastic/elasticsearch/blob/50793a68a8dfaca22ba1b552523baf4d5e2e15ea/server/src/main/java/org/elasticsearch/index/mapper/SourceValueFetcher.java#L107-L109) does not support `format` param and throws and exception if not null. 

So, I think we are good with this change

Reply:
That is already done in the `identity` method.